File: Two Pointers/1. Valid Palindrome/solution.scala
object Solution {
    def isPalindrome(s: String): Boolean = {
        var arr = s.split("").filter("""[a-zA-Z\d+]""".r.pattern.matcher(_).matches).map(_.toUpperCase)
        var left = 0
        var right = arr.length - 1
        while(left < right) {
            if(arr(left) != arr(right)) {return false}
            left += 1
            right -= 1
        }
        return true
    }
}

File: Two Pointers/1. Valid Palindrome/solution.rb
# @param {String} s
# @return {Boolean}
def is_palindrome(s)
  arr = s.split("").filter {|char| char =~ /[a-zA-Z\d+]/}.map {|char| char.upcase}

  left = 0
  right = arr.length - 1
  while left < right
    return false if arr[left] != arr[right]
    left += 1
    right -= 1
  end
  return true
end
